### PR TITLE
Content: Design Pattern - Device Monitoring

### LINKED ISSUE #2263: Content: Developing with FlowFuse - Device Monitoring
Part of the [Design Pattern/Developing with FlowFuse](#1797) series.

Article should detail the common use case of having several remote devices, and wanting to monitor them

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

Provides a Use Case, FlowFuse Ecosystem & Architecture breakdown for a "Device Monitoring" use case:

image

set in an Automative Plant:

image

## Related Issue(s)

Closes #2263

Art Request: https://github.com/FlowFuse/website/issues/2343

### CODE REVIEW

File: src/blog/2024/07/building-on-flowfuse-devices.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+This diagram also demonstrates a sample of the rich ecosystem of communication protocols that Node-RED can support:
+
+- **TCP:** Show here to interface between...
+- **EtherNet/IP:** Insert example of what this would be used for in such a use case...
+- **MQTT:**  Insert example of what this would be used for in such a use case...
+- **OPC-UA:**  Insert example of what this would be used for in such a use case...
+- **Modbus:**  Insert example of what this would be used for in such a use case...
+- **MC Protocol:**  Insert example of what this would be used for in such a use case...
+- **Siemens S7:**  Insert example of what this would be used for in such a use case...
```
Comment (joepavitt) [inline src/blog/2024/07/building-on-flowfuse-devices.md:75]: @Steve-Mcl would really appreciate a single line for each of these.

Comment (Steve-Mcl) [inline src/blog/2024/07/building-on-flowfuse-devices.md:76]:
```suggestion
- **TCP:** Read and process data from from bespoke and custom devices
- **EtherNet/IP:** Collect data from Allen Bradley and other EtherNet/IP devices
- **MQTT:**  Modern devices and factory generated IIoT data
- **OPC-UA:**  Communicate with newer PLCs and OPC Servers 
- **Modbus:**  Data collection from existing Modbus enabled devices like Temperature Probes, Invertors, Encoders
- **MC Protocol:**  Gather data from Mitsubishi PLCs
- **Siemens S7:**   Gather data from Siemens PLCs using the S7 Protocol
```

Not sure what angle to come at this with Joe - wrote something that can be iterated.

Comment (joepavitt) [reply src/blog/2024/07/building-on-flowfuse-devices.md:76]: Was going to try and tailor per the diagram we put together, and an example of how it'd fit into an automative plant

File: src/blog/2024/07/building-on-flowfuse-devices.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+---
+templateEngineOverride: njk, md
+title: "Building on FlowFuse: Remote Device Monitoring"
+subtitle: In this article we take a look at how elements of the FlowFuse ecosystem can be used to build powerful IoT applications for monitoring remote devices.
+description: In this article we take a look at how elements of the FlowFuse ecosystem can be used to build powerful IoT applications for monitoring remote devices.
+date: 2024-07-16
```
Comment (joepavitt) [inline src/blog/2024/07/building-on-flowfuse-devices.md:6]:
```suggestion
date: 2024-07-17
```

File: src/blog/2024/07/building-on-flowfuse-devices.md (hunk 1 above)
Comment (Steve-Mcl) [reply src/blog/2024/07/building-on-flowfuse-devices.md]: Most odd - I am 100% certain i already posted a comment/suggestion for these?
Am i missing something?

Comment (joepavitt) [reply src/blog/2024/07/building-on-flowfuse-devices.md]: you did - it was merged, but this comment wasn't marked as "resolved"

Comment (Steve-Mcl) [reply src/blog/2024/07/building-on-flowfuse-devices.md]: i did - here: https://github.com/FlowFuse/website/pull/2264#discussion_r1675968165

Is that not what you are after Joe?

PS, I note that the list includes "MC Protocol" and excludes "FINS"  but the diagram is the reverse.

Comment (Steve-Mcl) [reply src/blog/2024/07/building-on-flowfuse-devices.md]: ah ok. ta.

File: src/blog/2024/07/building-on-flowfuse-devices.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- **Cloud:** Here we demonstrate how a company may have external Cloud-based services they're dependent upon. For example, image analytics services. Node-RED can be used to parse machine imagery, and integrate straight with these external services from the FlowFuse server.
+
+This diagram also demonstrates a sample of the rich ecosystem of communication protocols that Node-RED can support:
+
+- **TCP:** Read and process data from from bespoke and custom devices
+- **EtherNet/IP:** Collect data from Allen Bradley and other EtherNet/IP devices
+- **MQTT:**  Modern devices and factory generated IIoT data
+- **OPC-UA:**  Communicate with newer PLCs and OPC Servers 
+- **Modbus:**  Data collection from existing Modbus enabled devices like Temperature Probes, Invertors, Encoders
+- **MC Protocol:**  Gather data from Mitsubishi PLCs
```
Comment (Steve-Mcl) [inline src/blog/2024/07/building-on-flowfuse-devices.md:75]:
```suggestion
- **MC Protocol:**  Gather data from Mitsubishi PLCs
- **FINS:**  Gather data from OMRON PLCs
```

Diagram has FINS but list did not.

**Review by Steve-Mcl — APPROVED** [review]
Approved baring 1 suggestion

**Pull request merged**